### DISCUSSION TITLE
[New Feature] Add multilingual syntax support to the parser

### DISCUSSION THREAD

**Discussion by prakanth97** [ballerina-platform/ballerina-lang#37012] [General]
This is to discuss and give updates on adding multilingual syntax support to the parser.

Issue: #36802 
Upstream branch: [multilingual-parser](https://github.com/ballerina-platform/ballerina-lang/tree/multilingual-parser)

`properties` files for each language(eg: keyword_en.properties) will be created where key is english translation keyword and value is corresponding encoded string of foreign language keywords

Changes has to be done in `BallerinaLexer` and `SyntaxKind` where `Keywords` has to be reflect based on the language. 

`Configurations` class to read properties file and get relevant foreign language keyword
```
public final class Configurations {

    private ResourceBundle properties = null;
    private static volatile Configurations instance = null;
    private static String language = "en";
    private static String country = "LK";

    private Configurations () {
        Locale currentLanguage;
        if (language == null || country == null) {
            currentLanguage = Locale.getDefault();
        } else {
            currentLanguage = new Locale(language, country);
        }
        this.properties = ResourceBundle.getBundle("token", currentLanguage);
    }

    public static Configurations getInstance() {
        if (instance == null) {
            synchronized (Configurations.class) {
                if (instance == null) {
                    instance = new Configurations();
                }
            }
        }
        return instance;
    }

    public String getProperty(String key){
        String result = null;
        if (key !=null){
            result = this.properties.getString(key);
            result = decodeTokenText(result);
        }
        return result;
    }

    public String decodeTokenText(String encodedTokenText) {
        return new String(Base64.getDecoder().decode(encodedTokenText));
    }
}
```

- Read foreign language keywords from properties file.
- Create enum using those keywords.
- Use that instead of english keywords.

**Comment by pcnfernando** [comment]
@prakanth97 Shall we update the thread with the changes to the approach?

Also, we need to decide on the below points.
1. naming convention for the property file
2. Where should this file reside in bal distro.
3. how to configure what language to be used for parsing.

**Reply by pcnfernando** [reply]
> @prakanth97 Shall we update the thread with the changes to the approach?
> 
> Also, we need to decide on the below points.
> 
> 1. naming convention for the property file
> 2. Where should this file reside in bal distro.
Maybe `<BAL_HOME>/resources` where `command-completions` reside.

> 3. how to configure what language to be used for parsing.
Maybe through System properties. This value can be a property read through the property file itself.
eg:
```
language = en

public = pub
.
.
.
```
